### PR TITLE
Example with resource bundles for the test harness

### LINKED ISSUE #5048: Add an example with resource bundles to the test harness
CocoaPods supports different possibilities for podspecs to declare resources. One of these are resource bundles. At installation time, CocoaPods generates an additional target for each resources bundles. The resources are added to that. At runtime, Xcode applies the built-in build rules to these files, just like it is done for resources in app or framework targets. The built product is a bundle directory, which holds the processed resources. This resources bundle has to be copied into the app bundle, which is done by the generated copy resources script.
The test suite defines via unit and integration specs how a working installation should look like. But this working installation is mostly based on human assumptions and reasoning. There are no specs currently which actually execute a copy resources script, which copies a resource bundle from a build directory into an app bundle. If any path around that is broken that might not be caught just by code review (as happened with #4954, #5028).
Our [examples](https://github.com/CocoaPods/CocoaPods/tree/master/examples) would be suitable for adding an example with a podspec, which uses resource bundles.
## Notes
- The examples are validated via the exit code of `xcodebuild`. This can be triggered via a Rake task which can be executed by [`rake examples:build`](https://github.com/CocoaPods/CocoaPods/blob/master/Rakefile#L244-L282).
- Only iOS and OS X examples are actually build and contribute in such a way to test coverage.
- Local podspecs / development pods as seen in the existing examples are preferable. Large downloads increase the execution time of the test suite, which makes them undesirable and should be avoided at all cost.
- Ideas for additional inspections / validations of the build products are welcome!

### PR DESCRIPTION
**Pull request by benasher44**
This fixes #5048 by adding some samples for testing different ways of adding resources to pods

### CODE REVIEW

**Comment by benasher44** [comment]
@mrackwitz so `rake examples:build` doesn't report any failures, but if I actually go to build my example project then, the build definitely fails as intended (because of #5034). Is there some other way these example projects are intended to be used, or should I fix up the `rake examples:build` to fail if `xcodebuild` fails?

**Comment by benasher44** [comment]
I should also mention that after doing all of this, this failure is just as easily discovered by doing `pod lib lint ResourcesBundleExample.podspec`. Knowing this, maybe there's a lighter-weight way to catch this in the test suite, given that `pod lib lint` basically does what I'm doing here (integrate a pod into a test project and try building)?

File: examples/Resource Bundle Example/Example Pods/Images.xcassets/Contents.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "info" : {
+    "version" : 1,
+    "author" : "xcode"
+  }
+}
```
Comment (segiddins) [inline examples/Resource Bundle Example/Example Pods/Images.xcassets/Contents.json]: trailing newlines on these files, please?

**Comment by segiddins** [comment]
This is awesome!

File: examples/Resource Bundle Example/Example Pods/Images.xcassets/Contents.json (hunk 1 above)
Comment (benasher44) [inline examples/Resource Bundle Example/Example Pods/Images.xcassets/Contents.json]: Sure I can do that. Xcode (7.2) generated these though, so if someone goes and modifies this later, you'll run into the same issue. Would you still like this fixed?

Comment (segiddins) [inline examples/Resource Bundle Example/Example Pods/Images.xcassets/Contents.json]: Not strictly necessary, but trailing newlines can help with diffing

Comment (benasher44) [inline examples/Resource Bundle Example/Example Pods/Images.xcassets/Contents.json]: Gotcha. Will fix!

**Comment by benasher44** [comment]
Done. Would still like to wait to see #5069 merged, so then I can update this PR and ensure the build fails with this change, since #5034 isn't fixed.

**Comment by benasher44** [comment]
I'm not sure what's going on with  `execution_output.txt`, but this has flaked on me a few times now.

**Comment by segiddins** [comment]
@benasher44 don't worry about it

**Comment by benasher44** [comment]
@mrackwitz travis is now failing in `xcodebuild`'s `CpResource` as expected.

**Comment by segiddins** [comment]
@mrackwitz if you could rebase your fix on top of this branch, I think we'll be good to merge?

**Comment by mrackwitz** [comment]
:white_check_mark: Done. That means, we can close this PR as the changes will be merged via #5048.

**Pull request closed without merging**